workflow/calc_weighted_score.cwl

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.0
+class: CommandLineTool
+baseCommand: python
+
+hints:
+  DockerRequirement:
+    dockerPull: python:3.7
+
+inputs:
+  - id: scores_file
+    type: File
+  - id: apollo
+    type: int
+  - id: outcome
+    type: int
+  - id: rembrandt
+    type: int
+  - id: roimasks
+    type: int
+
+arguments:
+  - valueFrom: calc_weighted_score.py
+  - valueFrom: $(inputs.scores_file)
+    prefix: -s
+  - valueFrom: $(inputs.apollo)
+    prefix: --apollo
+  - valueFrom: $(inputs.outcome)
+    prefix: --outcome
+  - valueFrom: $(inputs.rembrandt)
+    prefix: --rembrandt
+  - valueFrom: $(inputs.roimasks)
+    prefix: --roi
+  - valueFrom: scores.json
+    prefix: -r
+
+requirements:
+  - class: InlineJavascriptRequirement
+  - class: InitialWorkDirRequirement
+    listing:
+      - entryname: calc_weighted_score.py
+        entry: |
+          #!/usr/bin/env python
+          import argparse
+          import json
+
+          parser = argparse.ArgumentParser()
+          parser.add_argument("-s", "--scores", required=True)
+          parser.add_argument("-r", "--results", required=True)
+          parser.add_argument("--apollo", required=True, type=int)
+          parser.add_argument("--outcome", required=True, type=int)
+          parser.add_argument("--rembrandt", required=True, type=int)
+          parser.add_argument("--roi", required=True, type=int)
+
+          args = parser.parse_args()
+          with open(args.scores) as scores, open(args.results, "w") as out:
+            results = json.load(scores)
+            
+            apo_weight = results.get('APOLLO-2_score') * args.apollo
```

Comment:
Can it ever be `None`?  If theres ever not a score, this would fail.

Reply:
Hm good point. I was working under the assumption that there would always be a score.

Reply:
If there _is_ always a score, then it's not a big deal.

Reply:
No, I think it's better to be safe than sorry. Added some handling in case there is a null score.